build/Helix/ConvertWttLogToXUnit.cs

```diff
@@ -0,0 +1,321 @@
+                            elt.Element("Data").Element("WexContext").Value == "ModuleScope"
+                        );
+                };
+
+                Func<XElement, bool> isModuleOrClassScopeStart = (elt) =>
+                {
+                    return
+                        elt.Name == "Msg" &&
+                        elt.Element("Data") != null &&
+                        elt.Element("Data").Element("StartGroup") != null &&
+                        elt.Element("Data").Element("WexContext") != null &&
+                            (elt.Element("Data").Element("WexContext").Value == "ClassScope" ||
+                            elt.Element("Data").Element("WexContext").Value == "ModuleScope");
+                };
+
+                Func<XElement, bool> isModuleScopeEnd = (elt) =>
+                {
+                    return
+                        elt.Name == "Msg" &&
+                        elt.Element("Data") != null &&
+                        elt.Element("Data").Element("EndGroup") != null &&
+                        elt.Element("Data").Element("WexContext") != null &&
+                        elt.Element("Data").Element("WexContext").Value == "ModuleScope";
+                };
+
+                Func<XElement, bool> isClassScopeEnd = (elt) =>
+                {
+                    return
+                        elt.Name == "Msg" &&
+                        elt.Element("Data") != null &&
+                        elt.Element("Data").Element("EndGroup") != null &&
+                        elt.Element("Data").Element("WexContext") != null &&
+                        elt.Element("Data").Element("WexContext").Value == "ClassScope";
+                };
+
+                int testsExecuting = 0;
+                foreach (XElement element in doc.Root.Elements())
+                {
+                    // Capturing the frequency data to record accurate
+                    // timing data.
+                    if (element.Name == "RTI")
+                    {
+                        frequency = Int64.Parse(element.Attribute("Frequency").Value);
+                        string processName = element.Attribute("ProcessName").Value;
+                        jobId = processName.Split('\\')[3];
+                    }
+
+                    // It's possible for a test to launch another test. If that happens, we won't modify the
+                    // current result. Instead, we'll continue operating like normal and expect that we get two
+                    // EndTests nodes before our next StartTests. We'll check that we've actually got a stop time
+                    // before creating a new result. This will result in the two results being squashed
+                    // into one result of the outer test that ran the inner one.
+                    if (element.Name == "StartTest")
+                    {
+                        testsExecuting++;
+                        if (testsExecuting == 1)
```

Comment:
does this mean this can only nest one level deep?  Is that OK?

Reply:
from the comments I think no... It just means the results will be squashed into one completed test.